care/facility/api/viewsets/bed.py

```diff
@@ -240,3 +241,44 @@ def get_queryset(self):
             allowed_facilities = get_accessible_facilities(user)
             queryset = queryset.filter(bed__facility__id__in=allowed_facilities)
         return queryset
+
+    @extend_schema(
+        description="Toggle patient privacy",
+        responses={status.HTTP_200_OK: None},
+        request=None,
+        tags=["consultationbed"],
+    )
+    @action(detail=True, methods=["PATCH"])
+    def toggle_patient_privacy(self, request, external_id):
+        try:
+            user: User = request.user
+            consultation_bed: ConsultationBed = (
+                self.get_queryset().filter(external_id=external_id).first()
+            )
+
+            if consultation_bed and (
+                user.user_type == User.TYPE_VALUE_MAP["WardAdmin"]
+                or user.user_type == User.TYPE_VALUE_MAP["LocalBodyAdmin"]
+                or user.user_type == User.TYPE_VALUE_MAP["DistrictAdmin"]
+                or user.user_type == User.TYPE_VALUE_MAP["StateAdmin"]
+                or (
+                    user.user_type == User.TYPE_VALUE_MAP["Doctor"]
+                    and user.home_facility.external_id
+                    == consultation_bed.bed.facility.external_id
+                )
+                or (
+                    user.user_type == User.TYPE_VALUE_MAP["Staff"]
+                    and user.home_facility.external_id
+                    == consultation_bed.bed.facility.external_id
+                )
+            ):
+                consultation_bed.privacy = not consultation_bed.privacy
+                consultation_bed.save()
+                return Response({"status": "success"}, status=status.HTTP_200_OK)
+            raise PermissionDenied(
+                detail="You do not have permission to perform this action"
+            )
+        except PermissionDenied as e:
+            return Response({"message": e.detail}, status=status.HTTP_403_FORBIDDEN)
+        except Exception as e:
+            return Response({"message": str(e)}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
Why try catch the PermissionDenied exception? DRF would return 403 forbidden without explicitly having to try catch this right?

Same for 400 bad request.